test.js

```diff
@@ -89,3 +78,123 @@ test('all approved books have a slug defined in book-slugs.json', t => {
   });
   t.pass()
 })
+
+test('approved-book-list.json is JSON', t => {
+  const listData = fs.readFileSync(approvedBookListPath, { encoding: 'utf8' })
+  JSON.parse(listData)
+  t.pass()
+})
+
+test('approved-book-list.json matches schema', t => {
+  const listData = fs.readFileSync(approvedBookListPath, { encoding: 'utf8' })
+  const list = JSON.parse(listData)
+  const schemaData = fs.readFileSync(ablSchemaPath, { encoding: 'utf8' })
+  const schema = JSON.parse(schemaData)
+  const validate = new Ajv({ allErrors: true }).compile(schema)
+  const isValid = validate(list)
+  if (!isValid) { console.error(validate.errors) }
+  t.truthy(isValid)
+})
+
+test('approved-book-list.json approved-versions reference approved-books', t => {
+  const listData = fs.readFileSync(approvedBookListPath, { encoding: 'utf8' })
+  const list = JSON.parse(listData)
+  const validBookIds = list.approved_books.map(
+    entry => entry.collection_id != null ? entry.collection_id : entry.repository_name
+  )
+  list.approved_versions.forEach(entry => {
+    const bookId = entry.collection_id != null ? entry.collection_id : entry.repository_name
+    if (!validBookIds.includes(bookId)) {
+      t.fail(`Approved version references unknown book ${bookId}`)
+    }
+  });
+  t.pass()
+})
+
+test('approved-book-list.json collection_ids / repository names are unique', t => {
+  const listData = fs.readFileSync(approvedBookListPath, { encoding: 'utf8' })
+  const list = JSON.parse(listData)
+  list.approved_books.reduce((acc, entry) => {
+    const bookId = entry.collection_id != null ? entry.collection_id : entry.repository_name
+    if (acc.includes(bookId)) {
+      t.fail(`Duplicate book ID (collection or repository name) found: ${bookId}`)
+    }
+    acc.push(bookId)
+    return acc
+  }, [])
+  t.pass()
+})
+
+test('approved-book-list.json book UUIDs are unique', t => {
+  const listData = fs.readFileSync(approvedBookListPath, { encoding: 'utf8' })
+  const list = JSON.parse(listData)
+  const bookUUIDs = list.approved_books.map(
+    entry => entry.books.map(
+      book => book.uuid
+    )
+  ).flat()
+  bookUUIDs.reduce((acc, entry) =>{
+    if (acc.includes(entry)) {
+      t.fail(`Duplicate book UUID found: ${entry}`)
+    }
+    acc.push(entry)
+    return acc
+  }, [])
+  t.pass()
+})
+
```

Comment:
```suggestion

test('approved-book-list.json has at least one version for each book', t => {
  const listData = fs.readFileSync(approvedBookListPath, { encoding: 'utf8' })
  const abl = JSON.parse(listData)
  abl.approved_books.forEach(b => {
    if (b.repository_name) {
      t.truthy(abl.approved_versions.find(v => v.repository_name === b.repository_name), `No version for repository_name=${b.repository_name}`)
    } else {
      t.truthy(abl.approved_versions.find(v => v.collection_id === b.collection_id), `No version for collection_id=${b.collection_id}`)
    }
  })
})

```

It's unlikely to occur but since you added so many thorough tests here's one other corner case